src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents storage to store tasks
+ */
+public class Storage {
+
+    /** File path string to data of tasks to load/save */
+    private String filePath;
+
+    /**
+     * Storage constructor using file path to load/save tasks.
+     *
+     * @param filePath Path of file to save tasks.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the tasks in file path to a TaskList.
+     *
+     * @return TaskList (list of tasks).
+     * @throws DukeException If file is not found.
+     * Should not happen since existence of file is checked.
+     */
+    public TaskList load() throws DukeException {
```

Comment:
Again, I think the naming of this method could be made more explicit to demonstrate it's function, like: 
```suggestion
    public TaskList loadTasksFromStorage() throws DukeException {
```